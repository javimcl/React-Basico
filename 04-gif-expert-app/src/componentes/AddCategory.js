import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {

        setInputValue(e.target.value);
      //  console.log('llamado handle')

    }

    const handleSubmit = (e) => {
        e.preventDefault();
     //   console.log('submit')
        //const {setCategories} = props; 

        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }

        
       // console.log(setCategories);

    }

    return (

        <form onSubmit={ handleSubmit }>
            {/* <h1>{inputValue}</h1> */}
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}

            />
        </form>

    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}