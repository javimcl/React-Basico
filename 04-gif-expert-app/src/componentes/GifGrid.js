import React from 'react'
//import React, { useState, useEffect } from 'react'
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    //    const [count, setCount] = useState(0);

    const { data: images, loading } = useFetchGifs(category);
    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     // getGifs();
    //     getGifs(category)
    //         .then(setImages);
    //     // .then(imgs => setImages(imgs));
    // }, [category]);


    // const getGifs = async () => {
    //     const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI(category)}&limit=10&api_key=yQDsU2RIFRTYsbt5aZ37IG4b2hQftQlN`
    //     const resp = await fetch(url);
    //     const { data } = await resp.json();

    //     const gifs = data.map(img => {
    //         return {
    //             id: img.id,
    //             title: img.title,
    //             url: img.images?.downsized_medium.url
    //         }
    //     })

    //     console.log(gifs);
    //     setImages(gifs);

    // }

    //getGifs();

    return (
        <>
            <h3 className='animate__animated animate__swing'>{category}</h3>
            { loading && <p className='animate__animated animate__heartBeat'>Loading</p>}
            <div className='carGrid'>

                {/* <ol> */}
                {
                    images.map(img =>
                        <GifGridItem key={img.id} {...img} />
                        // <li key={id}>{title}</li>
                    )
                }
                {/* </ol> */}

                {/* <h3>{ count }</h3>
            <button onClick={() => setCount(count + 1)}></button> */}
            </div>
        </>
    )
}
