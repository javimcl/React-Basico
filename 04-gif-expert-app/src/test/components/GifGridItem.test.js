import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import { GifGridItem } from '../../componentes/GifGridItem'


describe('Prueba para <GifGridItem/>', () => {


    const title = 'Un titulo';
    const url = 'http';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);



    test('Debe mostrar el componente correctamente ', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe de tener la imagen igual al url y al de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe de tener animatr__fadeIn', () => {
        const div = wrapper.find('div');
        //console.log(div.props());
        //console.log(div.prop('className'));

        const className = div.prop('className');


        expect(className.includes('animate__fadeIn')).toBe(true);
    })



})
