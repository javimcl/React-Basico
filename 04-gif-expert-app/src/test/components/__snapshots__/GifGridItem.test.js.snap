// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para <GifGridItem/> Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="http"
  />
  <p>
    Un titulo
  </p>
</div>
`;
